src/VisualStudio/Core/Def/Implementation/Remote/RemoteHostClient.cs

```diff
@@ -25,26 +25,26 @@ protected RemoteHostClient(Workspace workspace)
 
         public event EventHandler<bool> ConnectionChanged;
 
-        public Task<Session> CreateCodeAnalysisServiceSessionAsync(Solution solution, CancellationToken cancellationToken)
+        public Task<Session> CreateServiceSessionAsync(string serviceName, Solution solution, CancellationToken cancellationToken)
```

Comment:
I will leave existing one here and add another API which existing one just delete with well known service.codeanalysis. or just add extension method on RemoteHostClient which has same shape as before.

some people want simplicity where they don't need to care about what service to give in.